From 74b74174d572fee4015b8f4e326db3cd949bcdc3 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Tue, 1 Dec 2020 14:57:39 +0100
Subject: [PATCH] lmw, lswi and related PowerPC insns aren't allowed on ppc64le

Newer binutils produce an error when the assembly contains lmw, stmw,
lswi, lswx, stswi, or stswx instructions in little-endian mode.

Only build and run the lsw and ldst_multiple testcases on ppc64[be].

https://bugs.kde.org/show_bug.cgi?id=427870
---
 NEWS                                  | 1 +
 none/tests/ppc64/Makefile.am          | 9 +++++++--
 none/tests/ppc64/ldst_multiple.vgtest | 1 +
 3 files changed, 9 insertions(+), 2 deletions(-)

Index: valgrind-3.16.1/none/tests/ppc64/Makefile.am
===================================================================
--- valgrind-3.16.1.orig/none/tests/ppc64/Makefile.am
+++ valgrind-3.16.1/none/tests/ppc64/Makefile.am
@@ -54,16 +54,20 @@ EXTRA_DIST = \
 
 check_PROGRAMS = \
 	allexec \
-	lsw jm-insns round \
+	jm-insns round \
 	test_isa_2_06_part1 test_isa_2_06_part2 test_isa_2_06_part3 \
 	test_dfp1 test_dfp2 test_dfp3 test_dfp4 test_dfp5 \
 	test_isa_2_07_part1 test_isa_2_07_part2 \
 	test_isa_3_0 \
 	subnormal_test \
-	test_tm test_touch_tm ldst_multiple data-cache-instructions \
+	test_tm test_touch_tm data-cache-instructions \
 	power6_mf_gpr std_reg_imm \
 	twi_tdi tw_td power6_bcmp
 
+# lmw, stmw, lswi, lswx, stswi, stswx compile (and run) only on big endian.
+if VGCONF_PLATFORMS_INCLUDE_PPC64BE_LINUX
+check_PROGRAMS += lsw ldst_multiple
+endif
 
 AM_CFLAGS    += @FLAG_M64@
 AM_CXXFLAGS  += @FLAG_M64@
Index: valgrind-3.16.1/none/tests/ppc64/ldst_multiple.vgtest
===================================================================
--- valgrind-3.16.1.orig/none/tests/ppc64/ldst_multiple.vgtest
+++ valgrind-3.16.1/none/tests/ppc64/ldst_multiple.vgtest
@@ -1 +1,2 @@
+prereq: ../../../tests/is_ppc64_BE
 prog: ldst_multiple
